### DISCUSSION TITLE
spades.py --test ends with error: suggestions?

### DISCUSSION THREAD

**Discussion by jieunii87** [ablab/spades#903] [Q&A]
I'm going to try install spades, but, do you have any suggestions as to why spades.py --test is ending with an error on this machine?  Thank you!!! This is totally stock SPAdes log.

Command line: /Users/jieunkim/anaconda3/bin/spades.py	--test	

System information:
  SPAdes version: 3.15.2
  Python version: 3.9.7
  OS: macOS-10.16-x86_64-i386-64bit

Output dir: /Users/jieunkim/Desktop/L.P/spades_test
Mode: read error correction and assembling
Debug mode is turned OFF

Dataset parameters:
  Standard mode
  For multi-cell/isolate data we recommend to use '--isolate' option; for single-cell MDA data use '--sc'; for metagenomic data use '--meta'; for RNA-Seq use '--rna'.
  Reads:
    Library number: 1, library type: paired-end
      orientation: fr
      left reads: ['/Users/jieunkim/anaconda3/share/spades/test_dataset/ecoli_1K_1.fq.gz']
      right reads: ['/Users/jieunkim/anaconda3/share/spades/test_dataset/ecoli_1K_2.fq.gz']
      interlaced reads: not specified
      single reads: not specified
      merged reads: not specified
Read error correction parameters:
  Iterations: 1
  PHRED offset will be auto-detected
  Corrected reads will be compressed
Assembly parameters:
  k: automatic selection based on read length
  Repeat resolution is enabled
  Mismatch careful mode is turned OFF
  MismatchCorrector will be SKIPPED
  Coverage cutoff is turned OFF
Other parameters:
  Dir for temp files: /Users/jieunkim/Desktop/L.P/spades_test/tmp
  Threads: 16
  Memory limit (in Gb): 250

======= SPAdes pipeline started. Log can be found here: /Users/jieunkim/Desktop/L.P/spades_test/spades.log

/Users/jieunkim/anaconda3/share/spades/test_dataset/ecoli_1K_1.fq.gz: max reads length: 100
/Users/jieunkim/anaconda3/share/spades/test_dataset/ecoli_1K_2.fq.gz: max reads length: 100

Reads length: 100

===== Before start started. 

===== Read error correction started. 

===== Read error correction started. 

== Running: /Users/jieunkim/anaconda3/bin/spades-hammer /Users/jieunkim/Desktop/L.P/spades_test/corrected/configs/config.info

  0:00:00.000     1M / 6M    INFO    General                 (main.cpp                  :  75)   Starting BayesHammer, built from refs/heads/master, git revision 54bed604152fd271b68bdf7b78bfd0788ea9e431
  0:00:00.014     1M / 6M    INFO    General                 (main.cpp                  :  76)   Loading config from /Users/jieunkim/Desktop/L.P/spades_test/corrected/configs/config.info
  0:00:00.044     1M / 7M    INFO    General                 (main.cpp                  :  78)   Maximum # of threads to use (adjusted due to OMP capabilities): 8
  0:00:00.044     1M / 7M    ERROR   General                 (memory_limit.cpp          :  47)   setrlimit(2) call failed, errno = 22

== Error ==  system call for: "['/Users/jieunkim/anaconda3/bin/spades-hammer', '/Users/jieunkim/Desktop/L.P/spades_test/corrected/configs/config.info']" finished abnormally, OS return value: 22

**Comment by asl** [comment]
This is a known issue on Monterey: https://github.com/ablab/spades/issues/839
